expandybird/expansion/expansion.py

```diff
@@ -231,13 +233,13 @@ def ExpandTemplate(resource, imports, env, validate_schema=False):
     except schema_validation.ValidationErrors as e:
       raise ExpansionError(resource['name'], e.message)
 
-  if source_file.endswith('jinja'):
+  if path.endswith('jinja'):
     expanded_template = ExpandJinja(
-        source_file, imports[source_file], resource, imports)
-  elif source_file.endswith('py'):
+        source_file, imports[source_file]['content'], resource, imports)
```

Comment:
probably the more "correct" thing to do here is to create an ImportFile class with path and content as members.